### PR TITLE
Mi Band 2: Change distance unit

### PR DESCRIPTION
**Pull request by joserebelo**
Allow the user to change the distance unit displayed on the band.

Found it in Mi Fit settings, hadn't noticed it before. It's a simple setting / implementation.

### CODE REVIEW

**Comment by joserebelo** [comment]
@

**Comment by joserebelo** [comment]
I'm assuming this also works on the Bip. @ashimokawa could you confirm?

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/miband/MiBand2Coordinator.java [hunk 1]
```diff
@@ -47,6 +47,8 @@
 import nodomain.freeyourgadget.gadgetbridge.model.DeviceType;
 import nodomain.freeyourgadget.gadgetbridge.util.Prefs;
 
+import static nodomain.freeyourgadget.gadgetbridge.GBApplication.getContext;
```
Comment (ashimokawa) [inline app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/miband/MiBand2Coordinator.java]: Thats not good

**Comment by ashimokawa** [comment]
@joserebelo 
It did not work on the Bip when I last tried. That was why I did not commit similar code I wrote.

Anyway, I would prefer to consolidate the metric/imperial setting for all devices into one global setting. I know we have one for HPlus already and the pebble can also do that in theory.

I don't think it is important to have in configurable per-device(?)

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/miband/MiBand2Coordinator.java (hunk 1 above)
Comment (joserebelo) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/miband/MiBand2Coordinator.java]: Was just following the HPlusCoordinator, from where I based my code. Probably should have stuck to the MiBand2Coordinator style though.

**Comment by joserebelo** [comment]
@ashimokawa Yes, that makes sense, I see no reason to have it configurable per-device as well.

This PR should probably be closed then, and done properly as a global setting.

**Comment by protomors** [comment]
>Anyway, I would prefer to consolidate the metric/imperial setting for all devices into one global setting. I know we have one for HPlus already and the pebble can also do that in theory.

@ashimokawa Device I am working on also has the setting for displaying imperial/metric units. I just started implementing custom settings page for it. But if this setting will be made global (I too think that this is better way) - then I will wait for it.

File: app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/miband/MiBand2Coordinator.java (hunk 1 above)
Comment (ashimokawa) [reply app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/miband/MiBand2Coordinator.java]: please just call GBApplication.getContext() and remove the import :)

**Comment by ashimokawa** [comment]
@joserebelo 
Moved to global settings, will you adapt the PR? 
Also the name in the commit is strange, it says something about DND instead of switching measurement system..
Meanwhile switching to miles also works on Bip with Firmware 0.0.9.14, I just tested it :+1: 

@protomors 
Done, PR welcome ;)

**Comment by joserebelo** [comment]
@ashimokawa I think I messed up when I rebased it (might be mixed with my other PR), I will fix them tomorrow.

**Comment by ashimokawa** [comment]
@joserebelo 
@protomors 
Just pushed 6d8ffad55cecc1da3fd6bc9a04d60605070260cd , some stuff can be removed now from your PR (the preferece related stuff), and in onSendConfiguation() you have to replace the constant and use SettingsActivity.MEASUREMENT_SYSTEM

**Comment by ashimokawa** [comment]
Are you still planning to update the PR, or shall I do it?

**Comment by joserebelo** [comment]
I have been a bit busy, but I will finish this today.

**Comment by joserebelo** [comment]
@ashimokawa I have rebased and fixed the PR.

By the way, it seems there are some localization problems. The preference summary will be correctly translated when the SettingsActivity loads, but not when changed (it changes to the preference value itself, instead of the localized string).

**Comment by ashimokawa** [comment]
@joserebelo 
Thanks!

And thanks for the hint regarding the translation - I will look into into it.

**Pull request merged**